src/main/java/picard/illumina/ExtractIlluminaBarcodes.java

```diff
@@ -406,13 +440,27 @@ private File getBarcodeFile(final int tile) {
         if (BARCODE_FILE != null) {
             parseBarcodeFile(messages);
         } else {
+            final int numBarcodes = readStructure.sampleBarcodes.length();
             final Set<String> barcodes = new HashSet<>();
+
             for (final String barcode : BARCODE) {
                 if (barcodes.contains(barcode)) {
                     messages.add("Barcode " + barcode + " specified more than once.");
                 }
                 barcodes.add(barcode);
-                final BarcodeMetric metric = new BarcodeMetric(null, null, barcode, new String[]{barcode});
+                int barcodeNum = 0;
```

Comment:
If a BARCODE was specified and not a BARCODE_FILE the metric map was constructed differently. I normalized this.